src/main/java/com/example/bank/account/TransferService.java

```diff
@@ -0,0 +1,57 @@
+package com.example.bank.account;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+@Slf4j
+@Getter
+public class TransferService {
+    public static void main(String[] args){
+        Account account = new Account();
+        List<Account> twoMyBankAccounts = account.createTwoMyBankAccounts();
+
+        Account polishAccount = new Account();
+        List<Account> twoPolishAccounts = account.createTwoPolishAccounts();
+
+        Account foreignAccount = new Account();
+        List<Account> twoForeignAccounts = account.createForeignAccounts();
+
+        TransferService myTransfer = new TransferService();
+        myTransfer.internalAndExternalTransfer(twoMyBankAccounts.get(0),twoMyBankAccounts.get(1),BigDecimal.valueOf(100L));
+    }
+    //TODO to może być jakiś interfejs w sumie; metoda abstrakcyjna; niezależna od banku i waluty
+    boolean internalAndExternalTransfer(Account firstAccount, Account secondAccount, BigDecimal amount){
+        if (firstAccount.getBalance().compareTo(amount) < 0 &&
+                firstAccount.getCurrency().equals(secondAccount.getCurrency()) &&
+                firstAccount.getAccountNumber().substring(0,2).equals(secondAccount.getAccountNumber().substring(0,2))
+        ){
+            log.info("Wrong amount or currency.");
+            return true;
+        } else {
+            firstAccount.setBalance(firstAccount.getBalance().subtract(amount));
+            secondAccount.setBalance(secondAccount.getBalance().add(amount));
+            log.info("Bank transfer for " + amount + " " + firstAccount.getCurrency() + " has done. " +
```

Comment:
chyba pokazywalem Ci jak logowac ladniej niz konkatenowac stringa calego ? Bo to troche brzydko wyglada i prymitywnie tyle konkatenacji. Napisz sobie jakis logUtils ewentualnie jak nie znajdziesz sensownego sposobu budowania parametrow do logow.

Reply:
Z tym sobie już poradziłem w nowym branchu.